### PR TITLE
[Issue-660] Show segments query should not fail, if table name is case insensitive

### PR DESCRIPTION
**Pull request by ManoharVanam**
Refer : #660 
Show segments query should not fail, if table name is case insensitive

### CODE REVIEW

File: integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataretention/DataRetentionTestCase.scala [hunk 1]
```diff
@@ -96,7 +96,7 @@ class DataRetentionTestCase extends QueryTest with BeforeAndAfterAll {
 
   test("RetentionTest_withoutDelete") {
     checkAnswer(
-      sql("SELECT country, count(salary) AS amount FROM DataRetentionTable WHERE country" +
+      sql("SELECT country, count(salary) AS amount FROM dataretentionTable WHERE country" +
```
Comment (ravipesala) [inline integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataretention/DataRetentionTestCase.scala]: why you have changed to lower case in test case?Even if it is mixed case also supposed to work right?

**Comment by ravipesala** [comment]
Also the test cases inside `DataCompactionCardinalityBoundryTest` not running, it is same case sensitive issue. Check the issue https://github.com/HuaweiBigData/carbondata/issues/648 . Please make sure that also fixed in this PR.

File: integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataretention/DataRetentionTestCase.scala (hunk 1 above)
Comment (ManoharVanam) [inline integration/spark/src/test/scala/org/carbondata/spark/testsuite/dataretention/DataRetentionTestCase.scala]: Here , Iam checking mixed case : dataretentionTable

**Comment by ManoharVanam** [comment]
retest this please

**Comment by ravipesala** [comment]
LGTM

**Pull request merged**